app/models/miq_alert.rb

```diff
@@ -72,6 +73,14 @@ def set_responds_to_events
     self.responds_to_events = events unless events.nil?
   end
 
+  def validate_delay_next_evaluation
+    next_frequency = (options || {}).fetch_path(:notifications, :delay_next_evaluation)
+    evaluation_method = expression[:eval_method] if expression.kind_of?(Hash)
```

Comment:
The suffix `if` means `evaluation_method` may be uninitialized, relies on uninitialized vars being nil in Ruby (depends on textual order, works in this case).  Personally I would write it another way, but that's a matter of opinion, your call.
E.g. scope everything under `if expression.kind_of?(Hash) && expression[:eval_method] == "dwh_generic"` on top.